chatmaild/src/chatmaild/greeterbot.py

```diff
@@ -0,0 +1,132 @@
+import time
+
+import deltachat
+from deltachat.tracker import ConfigureFailed
+from time import sleep
+import tempfile
+import os
+import configargparse
+import pkg_resources
+import secrets
+
+from chatmaild.database import Database
+from chatmaild.config import read_config
+from chatmaild.newemail import ALPHANUMERIC_PUNCT, CONFIG_PATH
+
+PASSDB_PATH = "/home/vmail/passdb.sqlite"
+
+
+def setup_account(data_dir: str, debug: bool) -> deltachat.Account:
+    """Create a deltachat account with a given addr/password combination.
+
+    :param data_dir: the directory where the data(base) is stored.
+    :param debug: whether to show log messages for the account.
+    :return: the deltachat account object.
+    """
+    chatmail_config = read_config(CONFIG_PATH)
+    addr = "hello@" + chatmail_config.mail_domain
+
+    try:
+        os.mkdir(os.path.join(data_dir, addr))
+    except FileExistsError:
+        pass
+    db_path = os.path.join(data_dir, addr, "db.sqlite")
+
+    ac = deltachat.Account(db_path)
+    if debug:
+        ac.add_account_plugin(deltachat.events.FFIEventLogger(ac))
+
+    ac.set_config("mvbox_move", "0")
+    ac.set_config("sentbox_watch", "0")
+    ac.set_config("bot", "1")
+    ac.set_config("mdns_enabled", "0")
+
+    if not ac.is_configured():
+        cleartext_password = "".join(
+            secrets.choice(ALPHANUMERIC_PUNCT)
+            for _ in range(chatmail_config.password_min_length + 3)
+        )
```

Comment:
i suggest to use `chatmaild.newemail.create_newemail_dict()['password']` to not add more places where passwords are generated.